client/routes/router.js

```diff
@@ -28,10 +28,16 @@ FlowRouter.route('/', {
 		Tracker.autorun(function(c) {
 			if (FlowRouter.subsReady() === true) {
 				Meteor.defer(function() {
-					if (Meteor.user() && Meteor.user().defaultRoom) {
-						const room = Meteor.user().defaultRoom.split('/');
+					const user = Meteor.user();
+					if (user && user.defaultRoom) {
+						const room = user.defaultRoom.split('/');
 						FlowRouter.go(room[0], { name: room[1] }, FlowRouter.current().queryParams);
-					} else {
+					}
+					else if (user && user.settings && user.settings.preferences.saveLastVisitedRoom && user.lastVisitedRoom) {
```

Comment:
Where are you defining the user preference of `saveLastVisitedRoom`? I searched our code base and this is not defined anywhere. Also, I believe the value `lastVisitedRoom` will not be propagated to the user since it's not on the publication.